# Importing

In [1]:
import pandas as pd
import numpy as np
from IPython.display import display

# Read data

In [3]:
df_ChordFormulae=pd.read_excel("../datasets/allChordsAndFormulae.xlsx")

In [3]:
df_ChordFormulae

chordName   formula
0      Major     0,4,7
1      Minor     0,3,7
2        Aug     0,4,8
3        Dim     0,3,6
4      Fifth       0,7
..       ...       ...
76      Dim7   0,3,6,9
77       m#5     0,3,8
78     m#7#5  0,3,8,11
79      m7#5  0,3,8,11
80      Aug6   0,4,8,9

[81 rows x 2 columns]

# Analysis

## see duplicate formulae

In [4]:
grpby_df_formula=df_ChordFormulae.groupby('formula')

In [5]:
for index in grpby_df_formula.groups.values():
    if len(index)>1:
        display(df_ChordFormulae.iloc[index])

chordName formula
3        Dim   0,3,6
64       mb5   0,3,6

chordName  formula
70       tri  0,3,6,9
76      Dim7  0,3,6,9

chordName formula
72     Min#5   0,3,8
77       m#5   0,3,8

chordName   formula
78     m#7#5  0,3,8,11
79      m7#5  0,3,8,11

chordName   formula
27       7b5  0,4,6,10
58    Maj7b5  0,4,6,10
74    Dom7b5  0,4,6,10

chordName   formula
19         7  0,4,7,10
73      Dom7  0,4,7,10

chordName      formula
15      13b9  0,4,7,10,13
23     7#5b9  0,4,7,10,13
29       7b9  0,4,7,10,13

chordName         formula
12        13  0,4,7,10,14,21
34      9b13  0,4,7,10,14,21

chordName      formula
22     7#5#9  0,4,7,10,15
24       7#9  0,4,7,10,15

chordName  formula
16         6  0,4,7,9
71     Majb7  0,4,7,9

chordName   formula
21       7#5  0,4,8,10
75    Dom7#5  0,4,8,10

In [7]:
len(grpby_df_formula)

68

### Result

1) there are 68 entries with duplicate formulae

# Convert string formulae to list formulae

In [8]:
def convert_Sf_Lf(lst):
    '''
    info: takes a list of strings and converts it to list of integers
    ip: list of strings
    op: list of integeres
    '''
    rlst=[]
    for i in lst:
        rlst.append(int(i))
    return rlst

In [9]:
df_ChordFormulae['formulaList']=df_ChordFormulae['formula'].str.split(",").apply(convert_Sf_Lf)

In [10]:
df_ChordFormulae.head(3)

chordName formula formulaList
0     Major   0,4,7   [0, 4, 7]
1     Minor   0,3,7   [0, 3, 7]
2       Aug   0,4,8   [0, 4, 8]

# Make chords

In [11]:
baseNotes=["A","A#","B","C","C#","D","D#","E","F","F#","G","G#"]

In [11]:
def makeChord(baseNote,chordName):
    '''
    info: takes a base note and a chord name, returns the notes in the chord
    ip: a base note(string) and a chord name(string)
    op: a list of notes of the chord
    '''
    i=baseNotes.index(baseNote)
    formulaList=df_ChordFormulae.loc[df_ChordFormulae['chordName']==chordName,'formulaList'].values
    chordNotes=[]
    for j in formulaList[0]:
        chordNotes.append(baseNotes[(i+j)%len(baseNotes)])
    return chordNotes

In [24]:
temp=[1,2,6,7,8]
temp.index(7)==len(temp)

False

In [31]:
def makeChord_String(baseNote,chordName):
    '''
    info: takes a base note and a chord name, returns the notes in the chord
    ip: a base note(string) and a chord name(string)
    op: a string of notes of the chord
    '''
    i=baseNotes.index(baseNote)
    formulaList=df_ChordFormulae.loc[df_ChordFormulae['chordName']==chordName,'formulaList'].values
    chordNotes_string=""
    for j in formulaList[0]:
        sep="," if formulaList[0].index(j)!=len(formulaList[0])-1 else ""
        chordNotes_string=chordNotes_string+(baseNotes[(i+j)%len(baseNotes)])+sep
    return chordNotes_string

In [32]:
# make all chords of all base notes
lstIndex_note_chordName=[]
lstAllChordNotes=[]
for note in baseNotes:
    for chordName in df_ChordFormulae['chordName'].unique():
        chordNotes=makeChord_String(note,chordName)
        lstIndex_note_chordName.append((note,chordName))
        lstAllChordNotes.append(chordNotes)

dict1={'indexNoteChord':lstIndex_note_chordName,'ChordNotes':lstAllChordNotes}  
df_allChordNotes=pd.DataFrame(dict1)
display(df_allChordNotes.head(3))

indexNoteChord ChordNotes
0     (A, Major)     A,C#,E
1     (A, Minor)      A,C,E
2       (A, Aug)     A,C#,F

In [33]:
df_allChordNotes.to_csv('../datasets/df_chordsData.csv',index=False)

# Analysis

## see duplicate chords

In [17]:
def checkChordEquality(lst1,lst2):
    '''
    info: checks whether list1 is in list2 and vice versa,
          that way even if the lists are shiffled we will be able to compare them
          this is done by converting lists into sets and taking symmetric difference
    ip: 2 lists
    op: if both lists contain same elements then True, o/w False
    '''
    lst1=set(lst1)
    lst2=set(lst2)
    
    if len(lst1.symmetric_difference(lst2))==0:
        return True
    else:
        return False

### test functions

#### checkChordEquality()

In [18]:
lst3=['A', 'C#', 'E']
lst4=['C#', 'E','A','B']
lst5=['A', 'C#', 'E','A']
lst6=['A', 'C#','A']
lst7=['C#','A','E']

In [19]:
checkChordEquality(lst3,lst5)

True

In [20]:
checkChordEquality(lst3,lst4)

False

In [21]:
checkChordEquality(lst3,lst6)

False

In [22]:
checkChordEquality(lst3,lst7)

True

### make groups of similar chords

In [23]:
df_allChordNotes.head()

indexNoteChord  ChordNotes
0     (A, Major)  [A, C#, E]
1     (A, Minor)   [A, C, E]
2       (A, Aug)  [A, C#, F]
3       (A, Dim)  [A, C, D#]
4     (A, Fifth)      [A, E]

In [24]:
i=1
df_allChordNotes.loc[i,'ChordNotes']

['A', 'C', 'E']

In [25]:
dict_similarChords={}
doneIndexes=[]
for i in range(len(df_allChordNotes.index)):
    if i not in doneIndexes:
        groupElements=[]
        for j in range(i,len(df_allChordNotes.index)):
            lst1=df_allChordNotes.loc[i,'ChordNotes']
            lst2=df_allChordNotes.loc[j,'ChordNotes']
            if checkChordEquality(lst1,lst2):
                doneIndexes.append(j)
                groupElements.append(j)
        dict_similarChords.update({df_allChordNotes.loc[i,'indexNoteChord']:groupElements})

In [26]:
len(dict_similarChords)

577

In [27]:
dict_similarChords

{('A', 'Major'): [0, 396, 401],
 ('A', 'Minor'): [1, 605],
 ('A', 'Aug'): [2, 326, 650],
 ('A', 'Dim'): [3, 64],
 ('A', 'Fifth'): [4, 6],
 ('A', 'Fifth 9th'): [5, 8, 576],
 ('A', 'Octave'): [7],
 ('A', 'Sus4'): [9, 410, 413],
 ('A', 11): [10],
 ('A', '11b9'): [11],
 ('A', 13): [12, 34],
 ('A', '13#9'): [13],
 ('A', '13b5b9'): [14],
 ('A', '13b9'): [15, 23, 29],
 ('A', 6): [16, 71, 772],
 ('A', '6add9'): [17, 198, 773],
 ('A', '6sus4'): [18, 442],
 ('A', 7): [19, 73],
 ('A', '7#11'): [20, 514],
 ('A', '7#5'): [21, 75],
 ('A', '7#5#9'): [22, 24],
 ('A', '7add11'): [25, 467],
 ('A', '7add13'): [26, 776],
 ('A', '7b5'): [27, 58, 74, 513, 544, 560],
 ('A', '7b5b9'): [28, 506],
 ('A', '7sus4'): [30],
 ('A', 9): [31],
 ('A', '9#11'): [32],
 ('A', '9#5'): [33, 845],
 ('A', '9b5'): [35, 195],
 ('A', '9sus4'): [36, 611, 827],
 ('A', 'add9'): [37, 585],
 ('A', 'augsus4'): [38, 406],
 ('A', 'm11'): [39, 296],
 ('A', 'm13'): [40],
 ('A', 'm6'): [41, 775],
 ('A', 'm6add9'): [42],
 ('A', 'm7'): [43, 

In [28]:
df_allChordNotes.head()

indexNoteChord  ChordNotes
0     (A, Major)  [A, C#, E]
1     (A, Minor)   [A, C, E]
2       (A, Aug)  [A, C#, F]
3       (A, Dim)  [A, C, D#]
4     (A, Fifth)      [A, E]

In [29]:
for index in dict_similarChords.values():
    if len(index)>1:
        display(df_allChordNotes.iloc[index])

indexNoteChord  ChordNotes
0       (A, Major)  [A, C#, E]
396    (C#, Min#5)  [C#, E, A]
401      (C#, m#5)  [C#, E, A]

indexNoteChord ChordNotes
1       (A, Minor)  [A, C, E]
605   (E, augsus4)  [E, A, C]

indexNoteChord  ChordNotes
2         (A, Aug)  [A, C#, F]
326      (C#, Aug)  [C#, F, A]
650       (F, Aug)  [F, A, C#]

indexNoteChord  ChordNotes
3        (A, Dim)  [A, C, D#]
64       (A, mb5)  [A, C, D#]

indexNoteChord ChordNotes
4         (A, Fifth)     [A, E]
6  (A, Fifth Octave)  [A, E, A]

indexNoteChord ChordNotes
5    (A, Fifth 9th)  [A, E, B]
8         (A, Sus2)  [A, B, E]
576       (E, Sus4)  [E, A, B]

indexNoteChord ChordNotes
9         (A, Sus4)  [A, D, E]
410  (D, Fifth 9th)  [D, A, E]
413       (D, Sus2)  [D, E, A]

indexNoteChord            ChordNotes
12        (A, 13)  [A, C#, E, G, B, F#]
34      (A, 9b13)  [A, C#, E, G, B, F#]

indexNoteChord         ChordNotes
15      (A, 13b9)  [A, C#, E, G, A#]
23     (A, 7#5b9)  [A, C#, E, G, A#]
29       (A, 7b9)  [A, C#, E, G, A#]

indexNoteChord      ChordNotes
16          (A, 6)  [A, C#, E, F#]
71      (A, Majb7)  [A, C#, E, F#]
772       (F#, m7)  [F#, A, C#, E]

indexNoteChord         ChordNotes
17      (A, 6add9)  [A, C#, E, F#, B]
198     (B, 9sus4)  [B, E, F#, A, C#]
773  (F#, m7add11)  [F#, A, C#, E, B]

indexNoteChord     ChordNotes
18      (A, 6sus4)  [A, D, E, F#]
442      (D, add9)  [D, F#, A, E]

indexNoteChord     ChordNotes
19         (A, 7)  [A, C#, E, G]
73      (A, Dom7)  [A, C#, E, G]

indexNoteChord         ChordNotes
20       (A, 7#11)  [A, C#, E, G, D#]
514    (D#, 7b5b9)  [D#, G, A, C#, E]

indexNoteChord     ChordNotes
21       (A, 7#5)  [A, C#, F, G]
75    (A, Dom7#5)  [A, C#, F, G]

indexNoteChord        ChordNotes
22     (A, 7#5#9)  [A, C#, E, G, C]
24       (A, 7#9)  [A, C#, E, G, C]

indexNoteChord        ChordNotes
25     (A, 7add11)  [A, C#, E, G, D]
467  (D, Maj9sus4)  [D, G, A, C#, E]

indexNoteChord         ChordNotes
26     (A, 7add13)  [A, C#, E, G, F#]
776     (F#, m7b9)  [F#, A, C#, E, G]

indexNoteChord      ChordNotes
27        (A, 7b5)  [A, C#, D#, G]
58     (A, Maj7b5)  [A, C#, D#, G]
74     (A, Dom7b5)  [A, C#, D#, G]
513      (D#, 7b5)  [D#, G, A, C#]
544   (D#, Maj7b5)  [D#, G, A, C#]
560   (D#, Dom7b5)  [D#, G, A, C#]

indexNoteChord          ChordNotes
28      (A, 7b5b9)  [A, C#, D#, G, A#]
506     (D#, 7#11)  [D#, G, A#, C#, A]

indexNoteChord        ChordNotes
33        (A, 9#5)  [A, C#, F, G, B]
845       (G, 9b5)  [G, B, C#, F, A]

indexNoteChord         ChordNotes
35        (A, 9b5)  [A, C#, D#, G, B]
195       (B, 9#5)  [B, D#, G, A, C#]

indexNoteChord       ChordNotes
36      (A, 9sus4)  [A, D, E, G, B]
611   (E, m7add11)  [E, G, B, D, A]
827     (G, 6add9)  [G, B, D, E, A]

indexNoteChord     ChordNotes
37       (A, add9)  [A, C#, E, B]
585     (E, 6sus4)  [E, A, B, C#]

indexNoteChord ChordNotes
38    (A, augsus4)  [A, D, F]
406     (D, Minor)  [D, F, A]

indexNoteChord          ChordNotes
39        (A, m11)  [A, C, E, G, B, D]
296     (C, Maj13)  [C, E, G, B, D, A]

indexNoteChord     ChordNotes
41         (A, m6)  [A, C, E, F#]
775     (F#, m7b5)  [F#, A, C, E]

indexNoteChord    ChordNotes
43         (A, m7)  [A, C, E, G]
259         (C, 6)  [C, E, G, A]
314     (C, Majb7)  [C, E, G, A]

indexNoteChord       ChordNotes
44    (A, m7add11)  [A, C, E, G, D]
260     (C, 6add9)  [C, E, G, A, D]
441     (D, 9sus4)  [D, G, A, C, E]

indexNoteChord     ChordNotes
46       (A, m7b5)  [A, C, D#, G]
284        (C, m6)  [C, D#, G, A]

indexNoteChord        ChordNotes
47       (A, m7b9)  [A, C, E, G, A#]
269    (C, 7add13)  [C, E, G, A#, A]

indexNoteChord       ChordNotes
48          (A, m9)  [A, C, E, G, B]
300  (C, Maj7add13)  [C, E, G, B, A]

indexNoteChord        ChordNotes
49          (A, m9b5)  [A, C, D#, G, B]
312  (C, m-Maj7add13)  [C, D#, G, B, A]

indexNoteChord             ChordNotes
53      (A, Maj13)  [A, C#, E, G#, B, F#]
768      (F#, m11)  [F#, A, C#, E, G#, B]

indexNoteChord          ChordNotes
57   (A, Maj7add13)  [A, C#, E, G#, F#]
777        (F#, m9)  [F#, A, C#, E, G#]

indexNoteChord        ChordNotes
62   (A, Maj9sus4)  [A, D, E, G#, B]
592    (E, 7add11)  [E, G#, B, D, A]

indexNoteChord     ChordNotes
67     (A, m-Maj7)  [A, C, E, G#]
323      (C, Aug6)  [C, E, G#, A]

indexNoteChord         ChordNotes
69   (A, m-Maj7add13)  [A, C, E, G#, F#]
778        (F#, m9b5)  [F#, A, C, E, G#]

indexNoteChord      ChordNotes
70        (A, tri)  [A, C, D#, F#]
76       (A, Dim7)  [A, C, D#, F#]
313       (C, tri)  [C, D#, F#, A]
319      (C, Dim7)  [C, D#, F#, A]
556      (D#, tri)  [D#, F#, A, C]
562     (D#, Dim7)  [D#, F#, A, C]
799      (F#, tri)  [F#, A, C, D#]
805     (F#, Dim7)  [F#, A, C, D#]

indexNoteChord ChordNotes
72      (A, Min#5)  [A, C, F]
77        (A, m#5)  [A, C, F]
648     (F, Major)  [F, A, C]

indexNoteChord     ChordNotes
78     (A, m#7#5)  [A, C, F, G#]
79      (A, m7#5)  [A, C, F, G#]

indexNoteChord      ChordNotes
80       (A, Aug6)  [A, C#, F, F#]
796   (F#, m-Maj7)  [F#, A, C#, F]

indexNoteChord  ChordNotes
81     (A#, Major)  [A#, D, F]
477     (D, Min#5)  [D, F, A#]
482       (D, m#5)  [D, F, A#]

indexNoteChord   ChordNotes
82     (A#, Minor)  [A#, C#, F]
686   (F, augsus4)  [F, A#, C#]

indexNoteChord   ChordNotes
83       (A#, Aug)  [A#, D, F#]
407       (D, Aug)  [D, F#, A#]
731      (F#, Aug)  [F#, A#, D]

indexNoteChord   ChordNotes
84       (A#, Dim)  [A#, C#, E]
145      (A#, mb5)  [A#, C#, E]

indexNoteChord   ChordNotes
85         (A#, Fifth)      [A#, F]
87  (A#, Fifth Octave)  [A#, F, A#]

indexNoteChord  ChordNotes
86   (A#, Fifth 9th)  [A#, F, C]
89        (A#, Sus2)  [A#, C, F]
657        (F, Sus4)  [F, A#, C]

indexNoteChord   ChordNotes
90        (A#, Sus4)  [A#, D#, F]
491  (D#, Fifth 9th)  [D#, A#, F]
494       (D#, Sus2)  [D#, F, A#]

indexNoteChord            ChordNotes
93        (A#, 13)  [A#, D, F, G#, C, G]
115     (A#, 9b13)  [A#, D, F, G#, C, G]

indexNoteChord         ChordNotes
96      (A#, 13b9)  [A#, D, F, G#, B]
104    (A#, 7#5b9)  [A#, D, F, G#, B]
110      (A#, 7b9)  [A#, D, F, G#, B]

indexNoteChord     ChordNotes
97         (A#, 6)  [A#, D, F, G]
152    (A#, Majb7)  [A#, D, F, G]
853        (G, m7)  [G, A#, D, F]

indexNoteChord        ChordNotes
98     (A#, 6add9)  [A#, D, F, G, C]
279     (C, 9sus4)  [C, F, G, A#, D]
854   (G, m7add11)  [G, A#, D, F, C]

indexNoteChord      ChordNotes
99     (A#, 6sus4)  [A#, D#, F, G]
523     (D#, add9)  [D#, G, A#, F]

indexNoteChord      ChordNotes
100        (A#, 7)  [A#, D, F, G#]
154     (A#, Dom7)  [A#, D, F, G#]

indexNoteChord         ChordNotes
101     (A#, 7#11)  [A#, D, F, G#, E]
595     (E, 7b5b9)  [E, G#, A#, D, F]

indexNoteChord       ChordNotes
102      (A#, 7#5)  [A#, D, F#, G#]
156   (A#, Dom7#5)  [A#, D, F#, G#]

indexNoteChord          ChordNotes
103    (A#, 7#5#9)  [A#, D, F, G#, C#]
105      (A#, 7#9)  [A#, D, F, G#, C#]

indexNoteChord          ChordNotes
106    (A#, 7add11)  [A#, D, F, G#, D#]
548  (D#, Maj9sus4)  [D#, G#, A#, D, F]

indexNoteChord         ChordNotes
107   (A#, 7add13)  [A#, D, F, G#, G]
857      (G, m7b9)  [G, A#, D, F, G#]

indexNoteChord      ChordNotes
108      (A#, 7b5)  [A#, D, E, G#]
139   (A#, Maj7b5)  [A#, D, E, G#]
155   (A#, Dom7b5)  [A#, D, E, G#]
594       (E, 7b5)  [E, G#, A#, D]
625    (E, Maj7b5)  [E, G#, A#, D]
641    (E, Dom7b5)  [E, G#, A#, D]

indexNoteChord         ChordNotes
109    (A#, 7b5b9)  [A#, D, E, G#, B]
587      (E, 7#11)  [E, G#, B, D, A#]

indexNoteChord          ChordNotes
114      (A#, 9#5)  [A#, D, F#, G#, C]
926      (G#, 9b5)  [G#, C, D, F#, A#]

indexNoteChord         ChordNotes
116      (A#, 9b5)  [A#, D, E, G#, C]
276       (C, 9#5)  [C, E, G#, A#, D]

indexNoteChord          ChordNotes
117    (A#, 9sus4)  [A#, D#, F, G#, C]
692   (F, m7add11)  [F, G#, C, D#, A#]
908    (G#, 6add9)  [G#, C, D#, F, A#]

indexNoteChord     ChordNotes
118     (A#, add9)  [A#, D, F, C]
666     (F, 6sus4)  [F, A#, C, D]

indexNoteChord    ChordNotes
119  (A#, augsus4)  [A#, D#, F#]
487    (D#, Minor)  [D#, F#, A#]

indexNoteChord              ChordNotes
120      (A#, m11)  [A#, C#, F, G#, C, D#]
377    (C#, Maj13)  [C#, F, G#, C, D#, A#]

indexNoteChord      ChordNotes
122       (A#, m6)  [A#, C#, F, G]
856      (G, m7b5)  [G, A#, C#, F]

indexNoteChord       ChordNotes
124       (A#, m7)  [A#, C#, F, G#]
340        (C#, 6)  [C#, F, G#, A#]
395    (C#, Majb7)  [C#, F, G#, A#]

indexNoteChord           ChordNotes
125  (A#, m7add11)  [A#, C#, F, G#, D#]
341    (C#, 6add9)  [C#, F, G#, A#, D#]
522    (D#, 9sus4)  [D#, G#, A#, C#, F]

indexNoteChord       ChordNotes
127     (A#, m7b5)  [A#, C#, E, G#]
365       (C#, m6)  [C#, E, G#, A#]

indexNoteChord          ChordNotes
128     (A#, m7b9)  [A#, C#, F, G#, B]
350   (C#, 7add13)  [C#, F, G#, B, A#]

indexNoteChord          ChordNotes
129         (A#, m9)  [A#, C#, F, G#, C]
381  (C#, Maj7add13)  [C#, F, G#, C, A#]

indexNoteChord          ChordNotes
130         (A#, m9b5)  [A#, C#, E, G#, C]
393  (C#, m-Maj7add13)  [C#, E, G#, C, A#]

indexNoteChord           ChordNotes
134    (A#, Maj13)  [A#, D, F, A, C, G]
849       (G, m11)  [G, A#, D, F, A, C]

indexNoteChord        ChordNotes
138  (A#, Maj7add13)  [A#, D, F, A, G]
858          (G, m9)  [G, A#, D, F, A]

indexNoteChord         ChordNotes
143  (A#, Maj9sus4)  [A#, D#, F, A, C]
673     (F, 7add11)  [F, A, C, D#, A#]

indexNoteChord      ChordNotes
148   (A#, m-Maj7)  [A#, C#, F, A]
404     (C#, Aug6)  [C#, F, A, A#]

indexNoteChord         ChordNotes
150  (A#, m-Maj7add13)  [A#, C#, F, A, G]
859          (G, m9b5)  [G, A#, C#, F, A]

indexNoteChord      ChordNotes
151      (A#, tri)  [A#, C#, E, G]
157     (A#, Dim7)  [A#, C#, E, G]
394      (C#, tri)  [C#, E, G, A#]
400     (C#, Dim7)  [C#, E, G, A#]
637       (E, tri)  [E, G, A#, C#]
643      (E, Dim7)  [E, G, A#, C#]
880       (G, tri)  [G, A#, C#, E]
886      (G, Dim7)  [G, A#, C#, E]

indexNoteChord    ChordNotes
153    (A#, Min#5)  [A#, C#, F#]
158      (A#, m#5)  [A#, C#, F#]
729    (F#, Major)  [F#, A#, C#]

indexNoteChord       ChordNotes
159    (A#, m#7#5)  [A#, C#, F#, A]
160     (A#, m7#5)  [A#, C#, F#, A]

indexNoteChord      ChordNotes
161     (A#, Aug6)  [A#, D, F#, G]
877    (G, m-Maj7)  [G, A#, D, F#]

indexNoteChord   ChordNotes
162     (B, Major)  [B, D#, F#]
558    (D#, Min#5)  [D#, F#, B]
563      (D#, m#5)  [D#, F#, B]

indexNoteChord  ChordNotes
163     (B, Minor)  [B, D, F#]
767  (F#, augsus4)  [F#, B, D]

indexNoteChord  ChordNotes
164       (B, Aug)  [B, D#, G]
488      (D#, Aug)  [D#, G, B]
812       (G, Aug)  [G, B, D#]

indexNoteChord ChordNotes
165       (B, Dim)  [B, D, F]
226       (B, mb5)  [B, D, F]

indexNoteChord  ChordNotes
166         (B, Fifth)     [B, F#]
168  (B, Fifth Octave)  [B, F#, B]

indexNoteChord   ChordNotes
167  (B, Fifth 9th)  [B, F#, C#]
170       (B, Sus2)  [B, C#, F#]
738      (F#, Sus4)  [F#, B, C#]

indexNoteChord  ChordNotes
171       (B, Sus4)  [B, E, F#]
572  (E, Fifth 9th)  [E, B, F#]
575       (E, Sus2)  [E, F#, B]

indexNoteChord              ChordNotes
174        (B, 13)  [B, D#, F#, A, C#, G#]
196      (B, 9b13)  [B, D#, F#, A, C#, G#]

indexNoteChord         ChordNotes
177      (B, 13b9)  [B, D#, F#, A, C]
185     (B, 7#5b9)  [B, D#, F#, A, C]
191       (B, 7b9)  [B, D#, F#, A, C]

indexNoteChord       ChordNotes
178         (B, 6)  [B, D#, F#, G#]
233     (B, Majb7)  [B, D#, F#, G#]
934       (G#, m7)  [G#, B, D#, F#]

indexNoteChord           ChordNotes
179     (B, 6add9)  [B, D#, F#, G#, C#]
360    (C#, 9sus4)  [C#, F#, G#, B, D#]
935  (G#, m7add11)  [G#, B, D#, F#, C#]

indexNoteChord      ChordNotes
180     (B, 6sus4)  [B, E, F#, G#]
604      (E, add9)  [E, G#, B, F#]

indexNoteChord      ChordNotes
181         (B, 7)  [B, D#, F#, A]
235      (B, Dom7)  [B, D#, F#, A]

indexNoteChord         ChordNotes
182      (B, 7#11)  [B, D#, F#, A, F]
676     (F, 7b5b9)  [F, A, B, D#, F#]

indexNoteChord     ChordNotes
183       (B, 7#5)  [B, D#, G, A]
237    (B, Dom7#5)  [B, D#, G, A]

indexNoteChord         ChordNotes
184     (B, 7#5#9)  [B, D#, F#, A, D]
186       (B, 7#9)  [B, D#, F#, A, D]

indexNoteChord         ChordNotes
187    (B, 7add11)  [B, D#, F#, A, E]
629  (E, Maj9sus4)  [E, A, B, D#, F#]

indexNoteChord          ChordNotes
188    (B, 7add13)  [B, D#, F#, A, G#]
938     (G#, m7b9)  [G#, B, D#, F#, A]

indexNoteChord     ChordNotes
189       (B, 7b5)  [B, D#, F, A]
220    (B, Maj7b5)  [B, D#, F, A]
236    (B, Dom7b5)  [B, D#, F, A]
675       (F, 7b5)  [F, A, B, D#]
706    (F, Maj7b5)  [F, A, B, D#]
722    (F, Dom7b5)  [F, A, B, D#]

indexNoteChord        ChordNotes
190     (B, 7b5b9)  [B, D#, F, A, C]
668      (F, 7#11)  [F, A, C, D#, B]

indexNoteChord         ChordNotes
197       (B, 9b5)  [B, D#, F, A, C#]
357      (C#, 9#5)  [C#, F, A, B, D#]

indexNoteChord       ChordNotes
199      (B, add9)  [B, D#, F#, C#]
747    (F#, 6sus4)  [F#, B, C#, D#]

indexNoteChord ChordNotes
200   (B, augsus4)  [B, E, G]
568     (E, Minor)  [E, G, B]

indexNoteChord            ChordNotes
201       (B, m11)  [B, D, F#, A, C#, E]
458     (D, Maj13)  [D, F#, A, C#, E, B]

indexNoteChord      ChordNotes
203        (B, m6)  [B, D, F#, G#]
937     (G#, m7b5)  [G#, B, D, F#]

indexNoteChord     ChordNotes
205        (B, m7)  [B, D, F#, A]
421         (D, 6)  [D, F#, A, B]
476     (D, Majb7)  [D, F#, A, B]

indexNoteChord        ChordNotes
206   (B, m7add11)  [B, D, F#, A, E]
422     (D, 6add9)  [D, F#, A, B, E]
603     (E, 9sus4)  [E, A, B, D, F#]

indexNoteChord    ChordNotes
208      (B, m7b5)  [B, D, F, A]
446        (D, m6)  [D, F, A, B]

indexNoteChord        ChordNotes
209      (B, m7b9)  [B, D, F#, A, C]
431    (D, 7add13)  [D, F#, A, C, B]

indexNoteChord         ChordNotes
210         (B, m9)  [B, D, F#, A, C#]
462  (D, Maj7add13)  [D, F#, A, C#, B]

indexNoteChord        ChordNotes
211         (B, m9b5)  [B, D, F, A, C#]
474  (D, m-Maj7add13)  [D, F, A, C#, B]

indexNoteChord               ChordNotes
215     (B, Maj13)  [B, D#, F#, A#, C#, G#]
930      (G#, m11)  [G#, B, D#, F#, A#, C#]

indexNoteChord           ChordNotes
219  (B, Maj7add13)  [B, D#, F#, A#, G#]
939        (G#, m9)  [G#, B, D#, F#, A#]

indexNoteChord          ChordNotes
224  (B, Maj9sus4)  [B, E, F#, A#, C#]
754   (F#, 7add11)  [F#, A#, C#, E, B]

indexNoteChord      ChordNotes
229    (B, m-Maj7)  [B, D, F#, A#]
485      (D, Aug6)  [D, F#, A#, B]

indexNoteChord          ChordNotes
231  (B, m-Maj7add13)  [B, D, F#, A#, G#]
940        (G#, m9b5)  [G#, B, D, F#, A#]

indexNoteChord     ChordNotes
232       (B, tri)  [B, D, F, G#]
238      (B, Dim7)  [B, D, F, G#]
475       (D, tri)  [D, F, G#, B]
481      (D, Dim7)  [D, F, G#, B]
718       (F, tri)  [F, G#, B, D]
724      (F, Dim7)  [F, G#, B, D]
961      (G#, tri)  [G#, B, D, F]
967     (G#, Dim7)  [G#, B, D, F]

indexNoteChord ChordNotes
234     (B, Min#5)  [B, D, G]
239       (B, m#5)  [B, D, G]
810     (G, Major)  [G, B, D]

indexNoteChord     ChordNotes
240     (B, m#7#5)  [B, D, G, A#]
241      (B, m7#5)  [B, D, G, A#]

indexNoteChord      ChordNotes
242      (B, Aug6)  [B, D#, G, G#]
958   (G#, m-Maj7)  [G#, B, D#, G]

indexNoteChord ChordNotes
243     (C, Major)  [C, E, G]
639     (E, Min#5)  [E, G, C]
644       (E, m#5)  [E, G, C]

indexNoteChord  ChordNotes
244     (C, Minor)  [C, D#, G]
848   (G, augsus4)  [G, C, D#]

indexNoteChord  ChordNotes
245       (C, Aug)  [C, E, G#]
569       (E, Aug)  [E, G#, C]
893      (G#, Aug)  [G#, C, E]

indexNoteChord   ChordNotes
246       (C, Dim)  [C, D#, F#]
307       (C, mb5)  [C, D#, F#]

indexNoteChord ChordNotes
247         (C, Fifth)     [C, G]
249  (C, Fifth Octave)  [C, G, C]

indexNoteChord ChordNotes
248  (C, Fifth 9th)  [C, G, D]
251       (C, Sus2)  [C, D, G]
819       (G, Sus4)  [G, C, D]

indexNoteChord ChordNotes
252       (C, Sus4)  [C, F, G]
653  (F, Fifth 9th)  [F, C, G]
656       (F, Sus2)  [F, G, C]

indexNoteChord           ChordNotes
255        (C, 13)  [C, E, G, A#, D, A]
277      (C, 9b13)  [C, E, G, A#, D, A]

indexNoteChord         ChordNotes
258      (C, 13b9)  [C, E, G, A#, C#]
266     (C, 7#5b9)  [C, E, G, A#, C#]
272       (C, 7b9)  [C, E, G, A#, C#]

indexNoteChord    ChordNotes
261     (C, 6sus4)  [C, F, G, A]
685      (F, add9)  [F, A, C, G]

indexNoteChord     ChordNotes
262         (C, 7)  [C, E, G, A#]
316      (C, Dom7)  [C, E, G, A#]

indexNoteChord         ChordNotes
263      (C, 7#11)  [C, E, G, A#, F#]
757    (F#, 7b5b9)  [F#, A#, C, E, G]

indexNoteChord      ChordNotes
264       (C, 7#5)  [C, E, G#, A#]
318    (C, Dom7#5)  [C, E, G#, A#]

indexNoteChord         ChordNotes
265     (C, 7#5#9)  [C, E, G, A#, D#]
267       (C, 7#9)  [C, E, G, A#, D#]

indexNoteChord        ChordNotes
268    (C, 7add11)  [C, E, G, A#, F]
710  (F, Maj9sus4)  [F, A#, C, E, G]

indexNoteChord      ChordNotes
270       (C, 7b5)  [C, E, F#, A#]
301    (C, Maj7b5)  [C, E, F#, A#]
317    (C, Dom7b5)  [C, E, F#, A#]
756      (F#, 7b5)  [F#, A#, C, E]
787   (F#, Maj7b5)  [F#, A#, C, E]
803   (F#, Dom7b5)  [F#, A#, C, E]

indexNoteChord          ChordNotes
271     (C, 7b5b9)  [C, E, F#, A#, C#]
749     (F#, 7#11)  [F#, A#, C#, E, C]

indexNoteChord         ChordNotes
278       (C, 9b5)  [C, E, F#, A#, D]
438       (D, 9#5)  [D, F#, A#, C, E]

indexNoteChord    ChordNotes
280      (C, add9)  [C, E, G, D]
828     (G, 6sus4)  [G, C, D, E]

indexNoteChord  ChordNotes
281   (C, augsus4)  [C, F, G#]
649     (F, Minor)  [F, G#, C]

indexNoteChord            ChordNotes
282       (C, m11)  [C, D#, G, A#, D, F]
539    (D#, Maj13)  [D#, G, A#, D, F, C]

indexNoteChord      ChordNotes
286        (C, m7)  [C, D#, G, A#]
502        (D#, 6)  [D#, G, A#, C]
557    (D#, Majb7)  [D#, G, A#, C]

indexNoteChord         ChordNotes
287   (C, m7add11)  [C, D#, G, A#, F]
503    (D#, 6add9)  [D#, G, A#, C, F]
684     (F, 9sus4)  [F, A#, C, D#, G]

indexNoteChord       ChordNotes
289      (C, m7b5)  [C, D#, F#, A#]
527       (D#, m6)  [D#, F#, A#, C]

indexNoteChord          ChordNotes
290      (C, m7b9)  [C, D#, G, A#, C#]
512   (D#, 7add13)  [D#, G, A#, C#, C]

indexNoteChord         ChordNotes
291          (C, m9)  [C, D#, G, A#, D]
543  (D#, Maj7add13)  [D#, G, A#, D, C]

indexNoteChord          ChordNotes
292          (C, m9b5)  [C, D#, F#, A#, D]
555  (D#, m-Maj7add13)  [D#, F#, A#, D, C]

indexNoteChord       ChordNotes
305  (C, Maj9sus4)  [C, F, G, B, D]
835    (G, 7add11)  [G, B, D, F, C]

indexNoteChord     ChordNotes
310    (C, m-Maj7)  [C, D#, G, B]
566     (D#, Aug6)  [D#, G, B, C]

indexNoteChord   ChordNotes
315     (C, Min#5)  [C, D#, G#]
320       (C, m#5)  [C, D#, G#]
891    (G#, Major)  [G#, C, D#]

indexNoteChord      ChordNotes
321     (C, m#7#5)  [C, D#, G#, B]
322      (C, m7#5)  [C, D#, G#, B]

indexNoteChord   ChordNotes
324    (C#, Major)  [C#, F, G#]
720     (F, Min#5)  [F, G#, C#]
725       (F, m#5)  [F, G#, C#]

indexNoteChord   ChordNotes
325    (C#, Minor)  [C#, E, G#]
929  (G#, augsus4)  [G#, C#, E]

indexNoteChord  ChordNotes
327      (C#, Dim)  [C#, E, G]
388      (C#, mb5)  [C#, E, G]

indexNoteChord    ChordNotes
328         (C#, Fifth)      [C#, G#]
330  (C#, Fifth Octave)  [C#, G#, C#]

indexNoteChord    ChordNotes
329  (C#, Fifth 9th)  [C#, G#, D#]
332       (C#, Sus2)  [C#, D#, G#]
900       (G#, Sus4)  [G#, C#, D#]

indexNoteChord    ChordNotes
333       (C#, Sus4)  [C#, F#, G#]
734  (F#, Fifth 9th)  [F#, C#, G#]
737       (F#, Sus2)  [F#, G#, C#]

indexNoteChord              ChordNotes
336       (C#, 13)  [C#, F, G#, B, D#, A#]
358     (C#, 9b13)  [C#, F, G#, B, D#, A#]

indexNoteChord         ChordNotes
339     (C#, 13b9)  [C#, F, G#, B, D]
347    (C#, 7#5b9)  [C#, F, G#, B, D]
353      (C#, 7b9)  [C#, F, G#, B, D]

indexNoteChord        ChordNotes
342    (C#, 6sus4)  [C#, F#, G#, A#]
766     (F#, add9)  [F#, A#, C#, G#]

indexNoteChord      ChordNotes
343        (C#, 7)  [C#, F, G#, B]
397     (C#, Dom7)  [C#, F, G#, B]

indexNoteChord         ChordNotes
344     (C#, 7#11)  [C#, F, G#, B, G]
838     (G, 7b5b9)  [G, B, C#, F, G#]

indexNoteChord     ChordNotes
345      (C#, 7#5)  [C#, F, A, B]
399   (C#, Dom7#5)  [C#, F, A, B]

indexNoteChord         ChordNotes
346    (C#, 7#5#9)  [C#, F, G#, B, E]
348      (C#, 7#9)  [C#, F, G#, B, E]

indexNoteChord          ChordNotes
349    (C#, 7add11)  [C#, F, G#, B, F#]
791  (F#, Maj9sus4)  [F#, B, C#, F, G#]

indexNoteChord     ChordNotes
351      (C#, 7b5)  [C#, F, G, B]
382   (C#, Maj7b5)  [C#, F, G, B]
398   (C#, Dom7b5)  [C#, F, G, B]
837       (G, 7b5)  [G, B, C#, F]
868    (G, Maj7b5)  [G, B, C#, F]
884    (G, Dom7b5)  [G, B, C#, F]

indexNoteChord        ChordNotes
352    (C#, 7b5b9)  [C#, F, G, B, D]
830      (G, 7#11)  [G, B, D, F, C#]

indexNoteChord         ChordNotes
359      (C#, 9b5)  [C#, F, G, B, D#]
519      (D#, 9#5)  [D#, G, B, C#, F]

indexNoteChord       ChordNotes
361     (C#, add9)  [C#, F, G#, D#]
909    (G#, 6sus4)  [G#, C#, D#, F]

indexNoteChord   ChordNotes
362  (C#, augsus4)  [C#, F#, A]
730    (F#, Minor)  [F#, A, C#]

indexNoteChord              ChordNotes
363      (C#, m11)  [C#, E, G#, B, D#, F#]
620     (E, Maj13)  [E, G#, B, D#, F#, C#]

indexNoteChord      ChordNotes
367       (C#, m7)  [C#, E, G#, B]
583         (E, 6)  [E, G#, B, C#]
638     (E, Majb7)  [E, G#, B, C#]

indexNoteChord          ChordNotes
368  (C#, m7add11)  [C#, E, G#, B, F#]
584     (E, 6add9)  [E, G#, B, C#, F#]
765    (F#, 9sus4)  [F#, B, C#, E, G#]

indexNoteChord     ChordNotes
370     (C#, m7b5)  [C#, E, G, B]
608        (E, m6)  [E, G, B, C#]

indexNoteChord         ChordNotes
371     (C#, m7b9)  [C#, E, G#, B, D]
593    (E, 7add13)  [E, G#, B, D, C#]

indexNoteChord          ChordNotes
372        (C#, m9)  [C#, E, G#, B, D#]
624  (E, Maj7add13)  [E, G#, B, D#, C#]

indexNoteChord         ChordNotes
373        (C#, m9b5)  [C#, E, G, B, D#]
636  (E, m-Maj7add13)  [E, G, B, D#, C#]

indexNoteChord           ChordNotes
386  (C#, Maj9sus4)  [C#, F#, G#, C, D#]
916    (G#, 7add11)  [G#, C, D#, F#, C#]

indexNoteChord      ChordNotes
391   (C#, m-Maj7)  [C#, E, G#, C]
647      (E, Aug6)  [E, G#, C, C#]

indexNoteChord     ChordNotes
402    (C#, m#7#5)  [C#, E, A, C]
403     (C#, m7#5)  [C#, E, A, C]

indexNoteChord  ChordNotes
405     (D, Major)  [D, F#, A]
801    (F#, Min#5)  [F#, A, D]
806      (F#, m#5)  [F#, A, D]

indexNoteChord  ChordNotes
408       (D, Dim)  [D, F, G#]
469       (D, mb5)  [D, F, G#]

indexNoteChord ChordNotes
409         (D, Fifth)     [D, A]
411  (D, Fifth Octave)  [D, A, D]

indexNoteChord ChordNotes
414       (D, Sus4)  [D, G, A]
815  (G, Fifth 9th)  [G, D, A]
818       (G, Sus2)  [G, A, D]

indexNoteChord           ChordNotes
417        (D, 13)  [D, F#, A, C, E, B]
439      (D, 9b13)  [D, F#, A, C, E, B]

indexNoteChord         ChordNotes
420      (D, 13b9)  [D, F#, A, C, D#]
428     (D, 7#5b9)  [D, F#, A, C, D#]
434       (D, 7b9)  [D, F#, A, C, D#]

indexNoteChord    ChordNotes
423     (D, 6sus4)  [D, G, A, B]
847      (G, add9)  [G, B, D, A]

indexNoteChord     ChordNotes
424         (D, 7)  [D, F#, A, C]
478      (D, Dom7)  [D, F#, A, C]

indexNoteChord         ChordNotes
425      (D, 7#11)  [D, F#, A, C, G#]
919    (G#, 7b5b9)  [G#, C, D, F#, A]

indexNoteChord      ChordNotes
426       (D, 7#5)  [D, F#, A#, C]
480    (D, Dom7#5)  [D, F#, A#, C]

indexNoteChord        ChordNotes
427     (D, 7#5#9)  [D, F#, A, C, F]
429       (D, 7#9)  [D, F#, A, C, F]

indexNoteChord        ChordNotes
430    (D, 7add11)  [D, F#, A, C, G]
872  (G, Maj9sus4)  [G, C, D, F#, A]

indexNoteChord      ChordNotes
432       (D, 7b5)  [D, F#, G#, C]
463    (D, Maj7b5)  [D, F#, G#, C]
479    (D, Dom7b5)  [D, F#, G#, C]
918      (G#, 7b5)  [G#, C, D, F#]
949   (G#, Maj7b5)  [G#, C, D, F#]
965   (G#, Dom7b5)  [G#, C, D, F#]

indexNoteChord          ChordNotes
433     (D, 7b5b9)  [D, F#, G#, C, D#]
911     (G#, 7#11)  [G#, C, D#, F#, D]

indexNoteChord         ChordNotes
440       (D, 9b5)  [D, F#, G#, C, E]
600       (E, 9#5)  [E, G#, C, D, F#]

indexNoteChord  ChordNotes
443   (D, augsus4)  [D, G, A#]
811     (G, Minor)  [G, A#, D]

indexNoteChord          ChordNotes
444       (D, m11)  [D, F, A, C, E, G]
701     (F, Maj13)  [F, A, C, E, G, D]

indexNoteChord    ChordNotes
448        (D, m7)  [D, F, A, C]
664         (F, 6)  [F, A, C, D]
719     (F, Majb7)  [F, A, C, D]

indexNoteChord       ChordNotes
449   (D, m7add11)  [D, F, A, C, G]
665     (F, 6add9)  [F, A, C, D, G]
846     (G, 9sus4)  [G, C, D, F, A]

indexNoteChord     ChordNotes
451      (D, m7b5)  [D, F, G#, C]
689        (F, m6)  [F, G#, C, D]

indexNoteChord        ChordNotes
452      (D, m7b9)  [D, F, A, C, D#]
674    (F, 7add13)  [F, A, C, D#, D]

indexNoteChord       ChordNotes
453         (D, m9)  [D, F, A, C, E]
705  (F, Maj7add13)  [F, A, C, E, D]

indexNoteChord        ChordNotes
454         (D, m9b5)  [D, F, G#, C, E]
717  (F, m-Maj7add13)  [F, G#, C, E, D]

indexNoteChord     ChordNotes
472    (D, m-Maj7)  [D, F, A, C#]
728      (F, Aug6)  [F, A, C#, D]

indexNoteChord      ChordNotes
483     (D, m#7#5)  [D, F, A#, C#]
484      (D, m7#5)  [D, F, A#, C#]

indexNoteChord   ChordNotes
486    (D#, Major)  [D#, G, A#]
882     (G, Min#5)  [G, A#, D#]
887       (G, m#5)  [G, A#, D#]

indexNoteChord   ChordNotes
489      (D#, Dim)  [D#, F#, A]
550      (D#, mb5)  [D#, F#, A]

indexNoteChord    ChordNotes
490         (D#, Fifth)      [D#, A#]
492  (D#, Fifth Octave)  [D#, A#, D#]

indexNoteChord    ChordNotes
495       (D#, Sus4)  [D#, G#, A#]
896  (G#, Fifth 9th)  [G#, D#, A#]
899       (G#, Sus2)  [G#, A#, D#]

indexNoteChord             ChordNotes
498       (D#, 13)  [D#, G, A#, C#, F, C]
520     (D#, 9b13)  [D#, G, A#, C#, F, C]

indexNoteChord          ChordNotes
501     (D#, 13b9)  [D#, G, A#, C#, E]
509    (D#, 7#5b9)  [D#, G, A#, C#, E]
515      (D#, 7b9)  [D#, G, A#, C#, E]

indexNoteChord       ChordNotes
504    (D#, 6sus4)  [D#, G#, A#, C]
928     (G#, add9)  [G#, C, D#, A#]

indexNoteChord       ChordNotes
505        (D#, 7)  [D#, G, A#, C#]
559     (D#, Dom7)  [D#, G, A#, C#]

indexNoteChord      ChordNotes
507      (D#, 7#5)  [D#, G, B, C#]
561   (D#, Dom7#5)  [D#, G, B, C#]

indexNoteChord           ChordNotes
508    (D#, 7#5#9)  [D#, G, A#, C#, F#]
510      (D#, 7#9)  [D#, G, A#, C#, F#]

indexNoteChord           ChordNotes
511    (D#, 7add11)  [D#, G, A#, C#, G#]
953  (G#, Maj9sus4)  [G#, C#, D#, G, A#]

indexNoteChord         ChordNotes
521      (D#, 9b5)  [D#, G, A, C#, F]
681       (F, 9#5)  [F, A, C#, D#, G]

indexNoteChord   ChordNotes
524  (D#, augsus4)  [D#, G#, B]
892    (G#, Minor)  [G#, B, D#]

indexNoteChord               ChordNotes
525      (D#, m11)  [D#, F#, A#, C#, F, G#]
782    (F#, Maj13)  [F#, A#, C#, F, G#, D#]

indexNoteChord        ChordNotes
529       (D#, m7)  [D#, F#, A#, C#]
745        (F#, 6)  [F#, A#, C#, D#]
800    (F#, Majb7)  [F#, A#, C#, D#]

indexNoteChord            ChordNotes
530  (D#, m7add11)  [D#, F#, A#, C#, G#]
746    (F#, 6add9)  [F#, A#, C#, D#, G#]
927    (G#, 9sus4)  [G#, C#, D#, F#, A#]

indexNoteChord       ChordNotes
532     (D#, m7b5)  [D#, F#, A, C#]
770       (F#, m6)  [F#, A, C#, D#]

indexNoteChord           ChordNotes
533     (D#, m7b9)  [D#, F#, A#, C#, E]
755   (F#, 7add13)  [F#, A#, C#, E, D#]

indexNoteChord           ChordNotes
534         (D#, m9)  [D#, F#, A#, C#, F]
786  (F#, Maj7add13)  [F#, A#, C#, F, D#]

indexNoteChord          ChordNotes
535         (D#, m9b5)  [D#, F#, A, C#, F]
798  (F#, m-Maj7add13)  [F#, A, C#, F, D#]

indexNoteChord       ChordNotes
553   (D#, m-Maj7)  [D#, F#, A#, D]
809     (F#, Aug6)  [F#, A#, D, D#]

indexNoteChord      ChordNotes
564    (D#, m#7#5)  [D#, F#, B, D]
565     (D#, m7#5)  [D#, F#, B, D]

indexNoteChord  ChordNotes
567     (E, Major)  [E, G#, B]
963    (G#, Min#5)  [G#, B, E]
968      (G#, m#5)  [G#, B, E]

indexNoteChord  ChordNotes
570       (E, Dim)  [E, G, A#]
631       (E, mb5)  [E, G, A#]

indexNoteChord ChordNotes
571         (E, Fifth)     [E, B]
573  (E, Fifth Octave)  [E, B, E]

indexNoteChord             ChordNotes
579        (E, 13)  [E, G#, B, D, F#, C#]
601      (E, 9b13)  [E, G#, B, D, F#, C#]

indexNoteChord        ChordNotes
582      (E, 13b9)  [E, G#, B, D, F]
590     (E, 7#5b9)  [E, G#, B, D, F]
596       (E, 7b9)  [E, G#, B, D, F]

indexNoteChord     ChordNotes
586         (E, 7)  [E, G#, B, D]
640      (E, Dom7)  [E, G#, B, D]

indexNoteChord     ChordNotes
588       (E, 7#5)  [E, G#, C, D]
642    (E, Dom7#5)  [E, G#, C, D]

indexNoteChord        ChordNotes
589     (E, 7#5#9)  [E, G#, B, D, G]
591       (E, 7#9)  [E, G#, B, D, G]

indexNoteChord          ChordNotes
602       (E, 9b5)  [E, G#, A#, D, F#]
762      (F#, 9#5)  [F#, A#, D, E, G#]

indexNoteChord           ChordNotes
606       (E, m11)  [E, G, B, D, F#, A]
863     (G, Maj13)  [G, B, D, F#, A, E]

indexNoteChord    ChordNotes
610        (E, m7)  [E, G, B, D]
826         (G, 6)  [G, B, D, E]
881     (G, Majb7)  [G, B, D, E]

indexNoteChord     ChordNotes
613      (E, m7b5)  [E, G, A#, D]
851        (G, m6)  [G, A#, D, E]

indexNoteChord       ChordNotes
614      (E, m7b9)  [E, G, B, D, F]
836    (G, 7add13)  [G, B, D, F, E]

indexNoteChord        ChordNotes
615         (E, m9)  [E, G, B, D, F#]
867  (G, Maj7add13)  [G, B, D, F#, E]

indexNoteChord         ChordNotes
616         (E, m9b5)  [E, G, A#, D, F#]
879  (G, m-Maj7add13)  [G, A#, D, F#, E]

indexNoteChord     ChordNotes
634    (E, m-Maj7)  [E, G, B, D#]
890      (G, Aug6)  [G, B, D#, E]

indexNoteChord     ChordNotes
645     (E, m#7#5)  [E, G, C, D#]
646      (E, m7#5)  [E, G, C, D#]

indexNoteChord  ChordNotes
651       (F, Dim)  [F, G#, B]
712       (F, mb5)  [F, G#, B]

indexNoteChord ChordNotes
652         (F, Fifth)     [F, C]
654  (F, Fifth Octave)  [F, C, F]

indexNoteChord           ChordNotes
660        (F, 13)  [F, A, C, D#, G, D]
682      (F, 9b13)  [F, A, C, D#, G, D]

indexNoteChord         ChordNotes
663      (F, 13b9)  [F, A, C, D#, F#]
671     (F, 7#5b9)  [F, A, C, D#, F#]
677       (F, 7b9)  [F, A, C, D#, F#]

indexNoteChord     ChordNotes
667         (F, 7)  [F, A, C, D#]
721      (F, Dom7)  [F, A, C, D#]

indexNoteChord      ChordNotes
669       (F, 7#5)  [F, A, C#, D#]
723    (F, Dom7#5)  [F, A, C#, D#]

indexNoteChord         ChordNotes
670     (F, 7#5#9)  [F, A, C, D#, G#]
672       (F, 7#9)  [F, A, C, D#, G#]

indexNoteChord        ChordNotes
683       (F, 9b5)  [F, A, B, D#, G]
843       (G, 9#5)  [G, B, D#, F, A]

indexNoteChord             ChordNotes
687       (F, m11)  [F, G#, C, D#, G, A#]
944    (G#, Maj13)  [G#, C, D#, G, A#, F]

indexNoteChord      ChordNotes
691        (F, m7)  [F, G#, C, D#]
907        (G#, 6)  [G#, C, D#, F]
962    (G#, Majb7)  [G#, C, D#, F]

indexNoteChord      ChordNotes
694      (F, m7b5)  [F, G#, B, D#]
932       (G#, m6)  [G#, B, D#, F]

indexNoteChord          ChordNotes
695      (F, m7b9)  [F, G#, C, D#, F#]
917   (G#, 7add13)  [G#, C, D#, F#, F]

indexNoteChord         ChordNotes
696          (F, m9)  [F, G#, C, D#, G]
948  (G#, Maj7add13)  [G#, C, D#, G, F]

indexNoteChord         ChordNotes
697          (F, m9b5)  [F, G#, B, D#, G]
960  (G#, m-Maj7add13)  [G#, B, D#, G, F]

indexNoteChord     ChordNotes
715    (F, m-Maj7)  [F, G#, C, E]
971     (G#, Aug6)  [G#, C, E, F]

indexNoteChord      ChordNotes
726     (F, m#7#5)  [F, G#, C#, E]
727      (F, m7#5)  [F, G#, C#, E]

indexNoteChord  ChordNotes
732      (F#, Dim)  [F#, A, C]
793      (F#, mb5)  [F#, A, C]

indexNoteChord    ChordNotes
733         (F#, Fifth)      [F#, C#]
735  (F#, Fifth Octave)  [F#, C#, F#]

indexNoteChord               ChordNotes
741       (F#, 13)  [F#, A#, C#, E, G#, D#]
763     (F#, 9b13)  [F#, A#, C#, E, G#, D#]

indexNoteChord          ChordNotes
744     (F#, 13b9)  [F#, A#, C#, E, G]
752    (F#, 7#5b9)  [F#, A#, C#, E, G]
758      (F#, 7b9)  [F#, A#, C#, E, G]

indexNoteChord       ChordNotes
748        (F#, 7)  [F#, A#, C#, E]
802     (F#, Dom7)  [F#, A#, C#, E]

indexNoteChord      ChordNotes
750      (F#, 7#5)  [F#, A#, D, E]
804   (F#, Dom7#5)  [F#, A#, D, E]

indexNoteChord          ChordNotes
751    (F#, 7#5#9)  [F#, A#, C#, E, A]
753      (F#, 7#9)  [F#, A#, C#, E, A]

indexNoteChord          ChordNotes
764      (F#, 9b5)  [F#, A#, C, E, G#]
924      (G#, 9#5)  [G#, C, E, F#, A#]

indexNoteChord     ChordNotes
807    (F#, m#7#5)  [F#, A, D, F]
808     (F#, m7#5)  [F#, A, D, F]

indexNoteChord   ChordNotes
813       (G, Dim)  [G, A#, C#]
874       (G, mb5)  [G, A#, C#]

indexNoteChord ChordNotes
814         (G, Fifth)     [G, D]
816  (G, Fifth Octave)  [G, D, G]

indexNoteChord          ChordNotes
822        (G, 13)  [G, B, D, F, A, E]
844      (G, 9b13)  [G, B, D, F, A, E]

indexNoteChord        ChordNotes
825      (G, 13b9)  [G, B, D, F, G#]
833     (G, 7#5b9)  [G, B, D, F, G#]
839       (G, 7b9)  [G, B, D, F, G#]

indexNoteChord    ChordNotes
829         (G, 7)  [G, B, D, F]
883      (G, Dom7)  [G, B, D, F]

indexNoteChord     ChordNotes
831       (G, 7#5)  [G, B, D#, F]
885    (G, Dom7#5)  [G, B, D#, F]

indexNoteChord        ChordNotes
832     (G, 7#5#9)  [G, B, D, F, A#]
834       (G, 7#9)  [G, B, D, F, A#]

indexNoteChord       ChordNotes
888     (G, m#7#5)  [G, A#, D#, F#]
889      (G, m7#5)  [G, A#, D#, F#]

indexNoteChord  ChordNotes
894      (G#, Dim)  [G#, B, D]
955      (G#, mb5)  [G#, B, D]

indexNoteChord    ChordNotes
895         (G#, Fifth)      [G#, D#]
897  (G#, Fifth Octave)  [G#, D#, G#]

indexNoteChord              ChordNotes
903       (G#, 13)  [G#, C, D#, F#, A#, F]
925     (G#, 9b13)  [G#, C, D#, F#, A#, F]

indexNoteChord          ChordNotes
906     (G#, 13b9)  [G#, C, D#, F#, A]
914    (G#, 7#5b9)  [G#, C, D#, F#, A]
920      (G#, 7b9)  [G#, C, D#, F#, A]

indexNoteChord       ChordNotes
910        (G#, 7)  [G#, C, D#, F#]
964     (G#, Dom7)  [G#, C, D#, F#]

indexNoteChord      ChordNotes
912      (G#, 7#5)  [G#, C, E, F#]
966   (G#, Dom7#5)  [G#, C, E, F#]

indexNoteChord          ChordNotes
913    (G#, 7#5#9)  [G#, C, D#, F#, B]
915      (G#, 7#9)  [G#, C, D#, F#, B]

indexNoteChord     ChordNotes
969    (G#, m#7#5)  [G#, B, E, G]
970     (G#, m7#5)  [G#, B, E, G]

In [30]:
len(dict_similarChords)

577

# Create Scale data

degrees in a scale i.e. notes in a scale are

1st degree = tonic

2nd degree = supertonic

3rd degree = mediant

4th degree = subdominant

5th degree = dominant

6th degree = submediant

7th degree = leading tone

8th degree = octave high of tonic

In [31]:
scale_formula_major=[0,2,4,5,7,9,11,12]
scale_formula_naturalMinor=[0,2,3,5,7,8,10,12]
scale_formula_harmonicMinor=[0,2,3,5,7,8,11,12]
scale_formula_melodicMinor=[0,2,3,5,7,9,11,12]
scale_formula_dorianb2=[0,1,3,5,7,9,11,12]
scale_formula_phrygingDominant=[0,1,4,5,7,8,11,12]
scale_formula_lydianMinor=[0,2,4,6,7,8,10,12]
scale_formula_mixolydianAugmented=[0,2,4,5,8,9,10,12]
scale_formula_locrianDominant=[0,2,4,5,6,8,10,12]

In [32]:
# create this dictinorary from a database later
dict_scaleName_formula={'Major':scale_formula_major,'Natural Minor':scale_formula_naturalMinor,\
                        'Harmonic Minor':scale_formula_harmonicMinor,\
                        'Melodic Minor':scale_formula_melodicMinor,'Dorianb2':scale_formula_dorianb2,\
                        'Phyryging Dominant':scale_formula_phrygingDominant,\
                        'Lydian Minor':scale_formula_lydianMinor,'Mixolydian Augmented':scale_formula_mixolydianAugmented,\
                        'Locrian Dominant':scale_formula_locrianDominant}

In [33]:
baseNotes

['A', 'A#', 'B', 'C', 'C#', 'D', 'D#', 'E', 'F', 'F#', 'G', 'G#']

In [34]:
class Scale():
    def __init__(self, name, baseNote):
        self.name=name
        self.baseNote=baseNote
        self.notes = self.populateNotes()
        
    def populateNotes(self):
        temp=[]
        for i in dict_scaleName_formula[self.name]:
            #print(i)
            index_baseNote=baseNotes.index(self.baseNote)
            temp.append(baseNotes[(index_baseNote+i)%len(baseNotes)])
        #print(temp)
        return temp

In [35]:
scale=Scale('Major','C')

In [36]:
temp=[1,2,3]

In [37]:
str(temp)

'[1, 2, 3]'

In [38]:
temp="[1,2,3]"

In [39]:
temp

'[1,2,3]'

In [40]:
temp.rstrip("]")

'[1,2,3'

In [41]:
temp.lstrip("[]")

'1,2,3]'

In [42]:
temp.strip("[").strip("]").split(",")

['1', '2', '3']

In [43]:
temp

'[1,2,3]'

In [44]:
def updateDb_scales():
    dict_scaleName_formula_db={'Note':[],'Scale':[],'Notes':[]}
    for baseNote in baseNotes:
        for name in dict_scaleName_formula.keys():
            dict_scaleName_formula_db['Note'].append(baseNote)
            dict_scaleName_formula_db['Scale'].append(name)
            dict_scaleName_formula_db['Notes'].append(str(Scale(name, baseNote).populateNotes()).strip("[").strip("]"))
    return dict_scaleName_formula_db

In [45]:
A = updateDb_scales()
import pandas as pd
df_scalesData=pd.DataFrame(A)

In [46]:
df_scalesData.head()

Note           Scale                                      Notes
0    A           Major  'A', 'B', 'C#', 'D', 'E', 'F#', 'G#', 'A'
1    A   Natural Minor     'A', 'B', 'C', 'D', 'E', 'F', 'G', 'A'
2    A  Harmonic Minor    'A', 'B', 'C', 'D', 'E', 'F', 'G#', 'A'
3    A   Melodic Minor   'A', 'B', 'C', 'D', 'E', 'F#', 'G#', 'A'
4    A        Dorianb2  'A', 'A#', 'C', 'D', 'E', 'F#', 'G#', 'A'

In [52]:
df_scalesData.to_csv('../datasets/df_scalesData.csv',index=False)

In [53]:
import pandas as pd
df_scaledData=pd.read_csv('../datasets/df_scalesData.csv')

In [54]:
df_scaledData.dtypes

Note     object
Scale    object
Notes    object
dtype: object

In [55]:
df_scaledData

Note                 Scale                                        Notes
0      A                 Major    'A', 'B', 'C#', 'D', 'E', 'F#', 'G#', 'A'
1      A         Natural Minor       'A', 'B', 'C', 'D', 'E', 'F', 'G', 'A'
2      A        Harmonic Minor      'A', 'B', 'C', 'D', 'E', 'F', 'G#', 'A'
3      A         Melodic Minor     'A', 'B', 'C', 'D', 'E', 'F#', 'G#', 'A'
4      A              Dorianb2    'A', 'A#', 'C', 'D', 'E', 'F#', 'G#', 'A'
..   ...                   ...                                          ...
103   G#              Dorianb2   'G#', 'A', 'B', 'C#', 'D#', 'F', 'G', 'G#'
104   G#    Phyryging Dominant   'G#', 'A', 'C', 'C#', 'D#', 'E', 'G', 'G#'
105   G#          Lydian Minor  'G#', 'A#', 'C', 'D', 'D#', 'E', 'F#', 'G#'
106   G#  Mixolydian Augmented  'G#', 'A#', 'C', 'C#', 'E', 'F', 'F#', 'G#'
107   G#      Locrian Dominant  'G#', 'A#', 'C', 'C#', 'D', 'E', 'F#', 'G#'

[108 rows x 3 columns]

In [51]:
df_scaledData['Notes'][0].strip("[").strip("]").split(", ")

["'A'", "'B'", "'C#'", "'D'", "'E'", "'F#'", "'G#'", "'A'"]

In [ ]:
# get simiilar scales

# Plan

1) create scales data

2) read it

3) make connection between chords and scales

4) maybe generate a data which contains
 * scale name
 * scale formula
 * all chords in it
 
5) find note similar scales

6) you have around 577 similar chords, unique. these are dimensions, or kinda "words", scqales are "sentences". Or some analogy like this.

7) 2 scales are chord similar if both of them contain same chords, ypu can have percentage similarity here

8) do this percentage thing for note similarity too.
 * make groups like differ by 1 note, differ by 2 notes, etc...
 
9) see if these 2 similarities are correlated

10) suggest chords based on these similarities

11) plans for mood:
    
 1. each scale has mood
 2. but mood depends on how you travel via chords
 3. mood depends on what chords you have recently played.
  * you cant say C major has positive vibes, because if you are in E major scale and suddenly you play C major, its gonna change mood